flusher.go

```diff
@@ -15,23 +16,32 @@ import (
 
 	"github.com/Sirupsen/logrus"
 	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/ssf"
+	"github.com/stripe/veneur/trace"
 )
 
 // Flush takes the slices of metrics, combines then and marshals them to json
 // for posting to Datadog.
 func (s *Server) Flush(interval time.Duration, metricLimit int) {
+	trace := trace.StartTrace("flush")
+	defer trace.Record("veneur.flush.trace", []*ssf.SSFTag{})
+
 	// right now we have only one destination plugin
 	// but eventually, this is where we would loop over our supported
 	// destinations
 	if s.IsLocal() {
-		s.FlushLocal(interval, metricLimit)
+		s.FlushLocal(trace.Attach(context.Background()), interval, metricLimit)
 	} else {
-		s.FlushGlobal(interval, metricLimit)
+		s.FlushGlobal(trace.Attach(context.Background()), interval, metricLimit)
```

Comment:
just wanna say i'm digging this api, i haven't read the code backing it yet, but from the outside it's quite nice